### DISCUSSION TITLE
v0.3.0 of the {surveydown} package is out! (sorry for the all breaking changes)

### DISCUSSION THREAD

**Discussion by jhelvy** [surveydown-dev/surveydown#97] [Announcements]
Hi everyone, 

We just released v0.3.0 of the {surveydown} package today. We made loads of breaking changes (apologies!), but these were necessary to make the design more secure and easier to use. Since the package is still in its very early stages, we hope we didn't inconvenience too many with these changes. We detailed the major changes in [this post](https://surveydown.org/blog/2024-09-18-new-app-design/). 

The biggest change is we are now adopting a two-file structure:

- `survey.qmd`: A Quarto document that contains the survey content (pages, questions, etc), which renders to an HTML file.
- `app.R`: An R script defining a Shiny app that contains the global settings (libraries, database configuration, etc.) and server configuration options (e.g., conditional skipping / display, etc.).

**These files must be named `survey.qmd` and `app.R`**

If you have any prior surveys, please check out the updated [documentation](https://surveydown.org/) and update your survey accordingly. First, update the package:

```r
remotes::install_github("surveydown-dev/surveydown", force = TRUE)
```

Then create a new template survey:

```r
surveydown::sd_create_survey("path/to/folder")
```

This will give you a template to work with. The biggest change is you will need to move your server logic out of the `survey.qmd` file and into the `app.R` file.

**Comment by willking98** [comment]
Nice one!
